devlab/tests/generate_load.py

```diff
@@ -942,9 +942,16 @@ def clean_networks(self):
         for network in self.neutronclient.list_networks()['networks']:
             if network['name'] not in nets_names:
                 continue
-            self.neutronclient.delete_network(self.get_net_id(network['name']))
+            net_id = self.get_net_id(network['name'])
+            self.clean_network_ports(net_id)
+            self.neutronclient.delete_network(net_id)
             print('Network "%s" has been deleted' % network['name'])
 
+    def clean_network_ports(self, net_id):
+        ports = self.neutronclient.list_ports(network_id=net_id)['ports']
```

Comment:
This will not work if port is allocated on a router with `neutron router-gateway-set` or `neutron router-interface-add`. For these ports you'd have to explicitly call `router-gateway-clear` and `router-interface-delete` respectively.